app.rb

```diff
@@ -0,0 +1,245 @@
+    def list_all_books
+      if @all_books.empty?
+        puts "No books available!\n\n"
+      else
+        @all_books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+        puts "\n\n"
+      end
+    end
+
+    def list_all_people
+      if @all_people.empty?
+        puts "No person info available!\n\n"
+      else
+        @all_people.each do |person|
+          puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+        end
+        puts "\n"
+      end
+    end
+
+    def list_books_with_options
+      @all_books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+      puts "\n"
+    end
+
+    def list_person_with_options
+      @all_people.each_with_index do |person, index|
+        puts "#{index}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+      puts "\n"
+    end
+
+    def list_person_rentals
+      if @all_rentals.empty?
+        puts "No rentals available!\n\n"
+      else
+        print 'ID of the person: '
+        person_id = gets.chomp
+        puts 'Rentals:'
+
+        selected_rentals = @all_rentals.select do |rental|
+          rental.person.id.to_i == person_id.to_i
+        end
+
+        if selected_rentals.empty?
+          puts "No rentals for this person!\n\n"
+        else
+          selected_rentals.each do |rental|
+            puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
+          end
+          puts "\n"
+        end
+      end
+    end
+  end
+
+  class App
+    include CreateEntity
+    include DisplayEntity
+
+    def initialize
+      @all_books = []
+      @all_people = []
+      @all_rentals = []
+    end
+
+    def run
+      loop do
+        print_menu
+        option = gets.chomp.to_i
+        puts "\n"
+        process_option(option)
+        break if option == 7
+      end
+    end
+
+    def process_option(option)
+      case option
+      when 1
+        list_all_books
+      when 2
+        list_all_people
+      when 3
+        create_person
+      when 4
+        create_book
+      when 5
+        create_rental
+      when 6
+        list_person_rentals
+      else
+        puts "Invalid option! Please choose a valid option.\n\n"
+      end
+    end
+
+    private
+
+    def save_new_people(person)
+      @all_people << person
+    end
+
+    def save_new_book(book)
+      @all_books << book
+    end
+
+    def save_rental(rental)
+      @all_rentals << rental
+    end
+
+    def print_menu
+      puts 'Welcome to School Library App!'
+      puts
+      puts 'Please choose an option by entering a number:'
+      puts '1) List all books'
+      puts '2) List all people'
+      puts '3) Create a person'
+      puts '4) Create a book'
+      puts '5) Create a rental'
+      puts '6) List rentals for a given person ID'
+      puts '7) Quit'
+      print 'Select an option: '
+    end
+  end
+
+  app = App.new
+  app.run
+end
```

Comment:
![image](https://github.com/SaneMyburg/school-library/assets/93039359/e130d0a1-62ed-4f74-b8ee-a451f5116c30)

- [ ] It would be nice if you follow the requirements and in the requirements, it is clearly mentioned that the user input and user input selection should be on the main.rb file according to the requirements which you can see in the image above ⬆️ Kindly make an main.rb file and put the highlighted code in that file 👍